tests/mock_cloud_sdk.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+from pasqal_cloud.job import CreateJob
+from pasqal_cloud.device import BaseConfig, EmulatorType
+from pasqal_cloud.batch import Batch
+from pasqal_cloud.utils.responses import PaginatedResponse
+
+from uuid import uuid4
+
+from unittest.mock import MagicMock
+from typing import Any
+
+
+class MockSDK:
+    """Helper class to mock the cloud SDK and skip the API calls.
+
+    Warning: This implements only a small subset of the functions available
+        in the cloud SDK; focusing on the functions used by qek.
+        This should be extended to support all methods and moved to the
+        pasqal-cloud repository so that it can be reused for all future
+        libraries using the SDK.
+
+    TODO:
+      - mock execution of jobs and batches (pending => running => done)
+      - set proper results for jobs (right now its entirely mocked)
+    """
+
+    def __init__(self) -> None:
+        self.batches: dict[str, Batch] = {}
+
+    def get_device_specs_dict(self) -> Any:
+        with open("tests/fixtures/device_specs.json", "r") as f:
+            return json.load(f)
+
+    def create_batch(
+        self,
+        serialized_sequence: str,
+        jobs: list[CreateJob],
+        open: bool | None = None,
```

Comment:
Why not defaulting to `False` ? Since later you call `bool(open)` which converts `None` to `False`.

Reply:
This method signature is copied directly from the SDK. For full context, in the SDK the default None value is used to be able to distinguish between the value being left unset by the user or set explicitly to False. (this is useful for backwards compatibility reasons)